### DISCUSSION TITLE
Bitwarden friendly password update example

### DISCUSSION THREAD

**Discussion by bilogic** [filamentphp/filament#8237] [Ideas]
1. The password manager Bitwarden's Chrome extension is able to detect most password changes and ask the user to confirm before it updates its records
2. However, it was not able to detect the password change on my custom page
    ```PHP
    protected function getHeaderActions(): array
    {
        return [
            Action::make('changePassword')
                ->form([
                    TextInput::make('new_password')
                        ->password()
                        ->label('New password')
                        ->required()
                        ->rule(Password::default()),
                    TextInput::make('new_password_confirmation')
                        ->password()
                        ->label('Confirm New password')
                        ->required()
                        ->same('new_password')
                        ->rule(Password::default()),
                ])
                ->action(function (array $data) {
                    auth()->user()->update([
                        'password' => Hash::make($data['new_password']),
                    ]);
                    Notification::make()
                        ->title('Password updated successfully')
                        ->icon('heroicon-o-document-text')
                        ->iconColor('success')
                        ->send();
                }),
        ];
    }
    ```
3. It will be helpful to have a user settings page that can change password as part of the documentation that is also friendly with Bitwarden

**Comment by pboivin** [comment]
Do you know what's missing to make this work with Bitwarden?

**Reply by bilogic** [reply]
Short answer, no. But I have used BW long enough to notice that simple POST requests are usually caught correctly, while complex AJAX won't be. If there is a way to adapt filament code to perform a simple POST, that should do the trick.

**Reply by danharrin** [reply]
Short answer, no. Bitwarden should be more intelligent or allow the use of attributes or classes to identify inputs
